hil/ext/switches/_dell_base.py

```diff
@@ -35,37 +36,37 @@ def disable_native(self, vlan_id):
         self.disable_vlan(vlan_id)
         self._sendline('sw trunk native vlan none')
 
+
     def get_port_networks(self, ports):
-        num_re = re.compile(r'(\d+)')
         port_configs = self._port_configs(ports)
-        result = {}
+        badchars = ' (Inactive)'
+        network_list = []
         for k, v in port_configs.iteritems():
-            native = v['Trunking Native Mode VLAN'].strip()
-            match = re.match(num_re, native)
-            if match:
-                # We need to call groups to get the part of the string that
-                # actually matched, because it could include some junk on the
-                # end, e.g. "100 (Inactive)".
-                num_str = match.groups()[0]
-                native = int(num_str)
+            non_natives = ''
+            non_native_list = []
+            # Get native vlan then remove junk if native not None
+            native_vlan = v['Trunking Native Mode VLAN'].strip()
+            if (native_vlan != 'none'):
+                native_vlan = ''.join(c for c in native_vlan
+                                      if c not in badchars)
```

Comment:
This is somewhat subjective, but I personally find python's if expression really hard to read. Maybe just use filter and a lambda?

Reply:
personally I find filter and lambdas hard to read. I find what @Izhmash has done to be more readable.